python/tests/test_ccb.py

```diff
@@ -122,3 +122,130 @@ def test_ccb_non_slot_none_outcome():
+            "gray",
+        ]
+
+        for i in range(1, n):
+            person = random.randint(0, 1)
+            chosen = [int(i) for i in np.random.permutation(2)]
+            rewards = [envs[offset][person][chosen[0]], envs[offset][person][chosen[1]]]
+
+            for i in range(len(rewards)):
+                rewards[i] += np.random.normal(0.5, variance)
+
+            current = {
+                "c": {
+                    "shared": {"name": people_ccb[person]},
+                    "_multi": [{"a": {"topic": topics_ccb[i]}} for i in range(2)],
+                    "_slots": [{"_id": i} for i in range(2)],
+                },
+                "_outcomes": [
+                    {
+                        "_label_cost": -min(rewards[i], 1),
+                        "_a": chosen[i:],
+                        "_p": [1.0 / (2 - i)] * (2 - i),
+                    }
+                    for i in range(2)
+                ],
+            }
+
+            current["c"]["shared"][random.choice(animals)] = random.random()
+            current["c"]["shared"][random.choice(animals)] = random.random()
+
+            current["c"]["_multi"][random.choice(range(len(current["c"]["_multi"])))][
+                random.choice(colors)
+            ] = random.random()
+            current["c"]["_multi"][random.choice(range(len(current["c"]["_multi"])))][
+                random.choice(colors)
+            ] = random.random()
+
+            current["c"]["_slots"][random.choice(range(len(current["c"]["_slots"])))][
+                random.choice(colors)
+            ] = random.random()
+            current["c"]["_slots"][random.choice(range(len(current["c"]["_slots"])))][
+                random.choice(colors)
+            ] = random.random()
+
+            yield current
+
+    def save_examples(examples, path):
+        with open(path, "w") as f:
+            for ex in examples:
+                f.write(f'{json.dumps(ex, separators=(",", ":"))}\n')
+
+    input_file = "ccb.json"
+    cache_dir = ".cache"
+    save_examples(
+        my_ccb_simulation(n=10, variance=0.1, bad_features=1, seed=0), input_file
+    )
+
+    assert os.path.exists(input_file)
+
+    vw = Vw(cache_dir, handler=None)
+    q = vw.train(
+        input_file, "-b 18 -q :: --ccb_explore_adf --dsjson", ["--invert_hash"]
+    )
+    automl = vw.train(
+        input_file,
+        "-b 20 --ccb_explore_adf --log_output stderr --dsjson --automl 4 --oracle_type one_diff --verbose_metrics",
+        ["--invert_hash", "--extra_metrics"],
+    )
+
+    automl_metrics = json.load(open(automl.outputs["--extra_metrics"][0]))
+
+    # we need champ switches to be zero for this to work
+    assert "total_champ_switches" in automl_metrics
+    assert automl_metrics["total_champ_switches"] == 0
+
+    fts_names_q = set([n for n in q[0].model9("--invert_hash").weights.index])
+    fts_names_automl = set(
+        [n for n in automl[0].model9("--invert_hash").weights.index if "[" not in n]
```

Comment:
leave comment that `[` implies a non-zero feature weight

Reply:
done